### ISSUE TITLE
[FEATURE REQ] Expose an overload with RequestContext in Azure.Data.Tables

### ISSUE THREAD

**Issue by lailabougria** [Azure/azure-sdk-for-net#32389]
### Library name

Azure.Data.Tables

### Please describe the feature.

Azure.Core offers the ability to customize error handling by passing a `RequestContext` specifying certain `ErrorOptions`, as documented in [this sample](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/RequestContext.md). We wanted to leverage this feature in Azure.Data.Tables to stop the SDK from throwing exceptions when performing data operations that we expect to fail sometimes, e.g. when entities are not found.

However, the APIs in Azure.Data.Tables don't offer the ability to pass in the `RequestContext` to leverage this feature. Unless I missed something, it is currently impossible to change the error behavior. 

Is this functionality in Azure.Core still supposed to be used in downstream libraries? If so, is it on the roadmap to incorporate it into downstream libraries such as Azure.Data.Tables?

**Comment by jsquire** [comment]
Thank you for your feedback.  Tagging and routing to the team member best able to assist.

**Comment by christothes** [comment]
Hi @lailabougria - Are there specific methods where you'd like to have control over the error behavior to prevent exceptions? I think we have already addressed most of them in recent releases.

**Comment by lailabougria** [comment]
An example is GetEntity(Async). We use this method while not necessarily expecting it to find the entity (null would be perfectly fine). Now it throws a RequestFailedException with an underlying 404. The 404 as a response is fine, but we're trying to avoid the Exception. 

I just went back to check the dependency version we're using, which is 12.6.1. The GetEntityIfExists method was added in 12.7.0, which seems to have been released 32 minutes ago 😅 

So I guess that solves this issue; thanks!
